File: Integer/Missing Number/By sorting.java
// Method - Sorting(BruteForce)
// Time Complexity T - O(n log n)
// Space Complexity  S - O(1) 


class Solution {
    public int missingNumber(int[] nums) {
        
        // sorting the array
        Arrays.sort(nums);
            
        // Ensure that "n" is at the last index of an array
        // if not then return the "n"
        if(nums[nums.length - 1] != nums.length){
            return nums.length;
        }
        
        // ensure that "0" is at the first index of an array
        // if not then return "0"
        else if(nums[0] != 0) {
            return 0;
        }
        // finding missing number between 0 to n-1
        else{
            
            // looping through 1 to n-1 element of an array 
            for(int i = 1; i < nums.length; i++){
                
                // accessing the previous value and add up "1" to match with the current value
                int expectedNum = nums[i-1] + 1;
                
                // if the current number is not equal to the expected number
                // then return the expected number as a missing number
                if(nums[i] != expectedNum){
                    return expectedNum;
                }
            }
        }
        // return when no missing values are there in the given array
        return -1;
    }
}

File: Array/Average by exclude min and max/By sorting.cpp
// Method - Sorting(Brute Force)
// Time Complexity = O(n)
// Space Complexity = O(1)

class Solution {
public:
    double average(vector<int>& salary) {
        
        // initializing the variable sum
        double sum;
        
        // sorting the given array
        sort(salary.begin(), salary.end());
        
        // looping from "1" to "n-2" element of the array
        // which is ignoring the first and last element of an array
        for(int i = 1; i < salary.size() - 1 ; i++){
            
            // sum up the array values
            sum = sum + salary[i];
        }
        
        // calculate the average by the obtained sum value and size of array minus 2
        // 2 is, since we have ignored minimum and maximum value from the array
        double avg = sum / (salary.size() - 2); 
        
        // return the average value
        return avg;
    }
};
